zipline/src/commonMain/kotlin/app/cash/zipline/ZiplineService.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline
+
+/**
+ * Implemented by all interfaces that can safely be passed from the host platform to Kotlin/JS or
+ * vice versa.
```

Comment:
When is an interface able to safely do so?

Reply:
I see, the `ZiplineServiceAdapter` is the important part here.

Is it because it's an interface that you don't wanna do like Wire's `Message`s ? Something like

`ZipineService<A: ZiplineServiceAdapter>` ?